### PR TITLE
ExploreOffers with tests

### LINKED ISSUE #198: (SP: 1) [Find offers] Implement search toolbar with title
UserStory: #67 
## Description
Add title with description, implement input with searching functionality to search for an offer by name, category and subject
## Design 
image
## Acceptance Criteria
- [ ] When a user clicks on the "Back to all categories", he/she should be redirected to "categories" page
- [ ] When a user selects category from category menu, all offers that are related to that category should be displayed
- [ ] When a user selects subject from subject menu, all offers that are related to that subject should be displayed
- [ ] When a user clicks on the "Search" button or press enter, search should be performed

### PR DESCRIPTION
**Pull request by OlhaOvsiienko**
image
image

### CODE REVIEW

**Review by YaroslavLys — APPROVED** [review]

**Review by YaroslavChuiko — APPROVED** [review]

**Review by holus-bolus — APPROVED** [review]

**Review by voloshynvv — COMMENTED** [review]
The state of category and subject **should not** be empty if we have appropriate value in the URL

I tried to correct comments.
image

**Comment by voloshkin-v** [comment]
`subjectId` is removed from `searchParams` when you select the category

https://github.com/Space2Study-UA-1156/Client-01/assets/56509285/28ddcb8f-723e-45e5-8051-b1f891ee2d9a

**Review by voloshynvv — COMMENTED** [review]
1) `subjectId` is removed from `searchParams` when you select the category (it is OK)
But when you remove the category, the subject should also be empty 

https://github.com/Space2Study-UA-1156/Client-01/assets/56509285/28ddcb8f-723e-45e5-8051-b1f891ee2d9a


2) Why did you decide to store the search keyword separately as `tutorLastName` and `tutorFirstName`? 
It would be better to keep it like `search=value`

The request will be sent with query `?search=user`

Example: 
Screenshot 2024-05-27 124914

3) There is no cross icon in the design
image

Actual result:
image

You can reset the key from URL if the input is empty when submitting the form

**Review by khytrykv — APPROVED** [review]

**Review by YaroslavChuiko — APPROVED** [review]

**Review by holus-bolus — APPROVED** [review]

**Comment by OlhaOvsiienko** [comment]
Search parameters for the field (tutor) are now set only when you click the Search or Enter button.

**Review by YaroslavChuiko — APPROVED** [review]

**Review by YaroslavLys — APPROVED** [review]

File: src/containers/explore-offers/ExploreOffers.jsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  useEffect(() => {
+    if (isMobile || isTablet) {
+      setSearch('')
+      const params = new URLSearchParams(searchParams.toString())
+      params.delete('search')
+      setSearchParams(params)
+    }
+  }, [isMobile, isTablet, setSearchParams, searchParams.toString()])
```
Comment (vlad-khrychov) [inline src/containers/explore-offers/ExploreOffers.jsx]:
```suggestion
  }, [isMobile, isTablet, setSearchParams, searchParams])
```

**Review by vlad-khrychov — APPROVED** [review]

**Pull request merged**